src/buffer.rs

```diff
@@ -35,60 +42,126 @@ impl AudioBuffer {
         }
     }
 
-    /// Number of channels in this AudioBuffer
+    /// Number of channels in this `AudioBuffer`
     pub fn number_of_channels(&self) -> usize {
         self.channels.len()
     }
 
-    /// Number of samples per channel in this AudioBuffer
-    pub fn sample_len(&self) -> usize {
+    /// Number of samples per channel in this `AudioBuffer`
+    pub fn length(&self) -> usize {
         self.channels.get(0).map(ChannelData::len).unwrap_or(0)
     }
 
-    /// Sample rate of this AudioBuffer in Hertz
+    /// Sample rate of this `AudioBuffer` in Hertz
     pub fn sample_rate(&self) -> SampleRate {
         self.sample_rate
     }
 
-    /// Duration in seconds of the AudioBuffer
+    /// Duration in seconds of the `AudioBuffer`
     pub fn duration(&self) -> f64 {
-        self.sample_len() as f64 / self.sample_rate.0 as f64
+        self.length() as f64 / self.sample_rate.0 as f64
+    }
+
+    /// Copy data from a given channel to the given `Vec`
+    pub fn copy_from_channel(&self, destination: &mut Vec<f32>, channel_number: usize) {
```

Comment:
destination can be `&mut [f32]` here, this allows for more flexibility (and I think, corresponds better to the spec). 
Same goes for `copy_from_channel_with_offset`

Reply:
Of course, this below no longer holds then
```
        // we use the `capacity` instead of `len` because destination is a write
        // buffer and we don't care of its current values (cf. `copy_to_channel`)```

But that would be a good thing. The caller should decide explicitely how many samples they want to receive